### ISSUE TITLE
Investigate failing e2e tests with Docker engine version 26, API version 1.45

### ISSUE THREAD

**Issue by lbeaufort** [fecgov/fecfile-web-app#1941]
Business Reason As a developer, I will be able to run end-to-end tests with the latest docker engine versions so that I can keep the underlying technology up to date.

### QA Notes ###
See successful E2E tests:
[https://app.circleci.com/pipelines/github/fecgov/fecfile-web-app/6199/workflows/aecaf364-6ea0-41d7-aee0-163f26a0ae5c/jobs/21335|https://app.circleci.com/pipelines/github/fecgov/fecfile-web-app/6199/workflows/aecaf364-6ea0-41d7-aee0-163f26a0ae5c/jobs/21335]

!image-20240808-131844.png|width=1077,height=142,alt="image-20240808-131844.png"!

!image-20240808-131953.png|width=1031,height=603,alt="image-20240808-131953.png"!

### DEV Notes ###

 
 On May 2, the e2e tests passed for the `release/sprint-41` branch. When I re-ran them today (5/15), they failed for the same commit: https://app.circleci.com/pipelines/github/fecgov/fecfile-web-app?branch=release%2Fsprint-41. One difference between the failing and passing tests:

On the `Setup a remote Docker engine` step:
**Passing**
```
Created container accessible with:
  DOCKER_HOST=unix:///var/run/docker.sock
  DOCKER_MACHINE_NAME=localhost
Server Engine Details:
  Version:          24.0.9
  API version:      1.43 (minimum version 1.12)
```
**Failing**
```
Created container accessible with:
  DOCKER_HOST=unix:///var/run/docker.sock
  DOCKER_MACHINE_NAME=localhost
Server Engine Details:
  Version:          26.0.2
  API version:      1.45 (minimum version 1.24)
```

Pinning the version to 24 in the Circle config resulted in [passing e2e tests](https://app.circleci.com/pipelines/github/fecgov/fecfile-web-app/5672/workflows/e53db6ad-04c0-42c9-910d-e03c5d2330c8/jobs/19601). The best guess I have so far is that the Docker API had a breaking change: https://docs.docker.com/engine/api/version-history/#v145-api-changes. Hotfix PRs in progress,

Also, I'd like to understand how we're using docker.io/jwilder/dockerize and if that's a factor. 

_Originally posted by @lbeaufort in https://github.com/fecgov/fecfile-web-app/issues/1928#issuecomment-2113503624_

NOTE: To speed up debugging, try limiting the number of e2e tests to one. Also, check for latest release version of Docker 26

## QA Notes ##
Passes QA if merge into develop runs successfully in nightly run.

### Design ###

null

 See full ticket and images here: [FECFILE-390](https://fecgov.atlassian.net/browse/FECFILE-390)

Pull Request: https://github.com/fecgov/fecfile-web-app/pull/2076

**Comment by AureliaKhorsand** [comment]
Blocked by #1925

**Comment by sasha-dresden** [comment]
I'm putting this back to Sprint Backlog. I'm not getting anywhere with this. I'm going to work on some other stuff and come back to this, but if someone else wants to take a crack at it, feel free.

What I've discovered/tried:
- The issue is definitely related to Docker v26. I tried with v25 and it succeeded, something about 26 is the issue.
- I tried switching from a docker executor to a machine build. I thought maybe it had something to do with the executor, but when the machine used docker 26, same issue.
- The issue appears to be that the docker container for the front-end can't communicate to the docker container for the backend.
  - It fails when cypress attempts to hit http://localhost:8080/api/v1/user/login/authenticate which is the first communication between the front and the back and just verifies if it's working.
  - However, I also attempted to curl that endpoint from within the docker bash shell and that worked. It's just when it's in cypress
  - It appears it never reaches the backend. I checked the container logs for the api and there was no error or any indication that cypress had even reached the backend.
- One of the biggest breaking changes in Docker 26 was that legacy API versions were no longer supported. They did provide a way to override this with an environment variable, which I set, to no avail.

I'm starting to suspect it has something to do with cypress and how it interacts with docker in this instance, so maybe there's some configuration there we can change.

**Comment by Elaine-Krauss-TCG** [comment]
`Also, I'd like to understand how we're using docker.io/jwilder/dockerize and if that's a factor.`

After doing a little digging and experimentation, we're using dockerize for the `-wait` and `-timeout` commands.  We use these commands to ensure that the API  docker container has fully started before starting e2e testing.  You can find the documentation for dockerize [here](https://hub.docker.com/r/jwilder/dockerize)
